h/api/storage.py

```diff
@@ -180,6 +180,46 @@ def update_document_metadata(session,
             **document_meta_dict)
 
 
+def update_annotation(session, annotation_id, data):
+    """
+    Update an existing annotation and its associated document metadata.
+
+    Update the annotation identified by annotation_id with the given
+    data. Create, delete and update document metadata as appropriate.
+
+    :param session: the database session
+    :type session: sqlalchemy.orm.session.Session
+
+    :param annotation_id: the ID of the annotation to be updated, this is
+        assumed to be a validated ID of an annotation that does already exist
+        in the database
+    :type annotation_id: string
+
+    :param data: the validated data with which to update the annotation
+    :type data: dict
+
+    :returns: the updated annotation
+    :rtype: h.api.models.Annotation
+
+    """
+    # Remove any 'document' field first so that we don't try to save it on the
+    # annotation object.
+    document = data.pop('document', None)
+
+    annotation = models.Annotation.query.get(annotation_id)
```

Comment:
This should go through the passed-in session: `session.query(models.Annotation).get(annotation_id)`